pvlib/pvsystem.py

```diff
@@ -944,75 +945,61 @@ def physicaliam(aoi, n=1.526, K=4., L=0.002):
     return iam
 
 
-def calcparams_desoto(poa_global, temp_cell, alpha_isc, module_parameters,
-                      EgRef, dEgdT, M=1, irrad_ref=1000, temp_ref=25):
+def calcparams_desoto(effective_irradiance, temp_cell,
+                      alpha_sc, a_ref, I_L_ref, I_o_ref, R_sh_ref, R_s, 
+                      EgRef=1.121, dEgdT=-0.0002677,
+                      irrad_ref=1000, temp_ref=25):
     '''
-    Applies the temperature and irradiance corrections to inputs for
-    singlediode.
-
-    Applies the temperature and irradiance corrections to the IL, I0,
-    Rs, Rsh, and a parameters at reference conditions (IL_ref, I0_ref,
-    etc.) according to the De Soto et. al description given in [1]. The
-    results of this correction procedure may be used in a single diode
-    model to determine IV curves at irradiance = S, cell temperature =
-    Tcell.
+    Calculates five parameter values for the single diode equation at 
+    effective irradiance and cell temperature using the De Soto et al. 
+    model described in [1]. The five values returned by calcparams_desoto
+    can be used by singlediode to calculate an IV curve.
 
     Parameters
     ----------
-    poa_global : numeric
-        The irradiance (in W/m^2) absorbed by the module.
+    effective_irradiance : numeric
+        The irradiance (W/m2) that is converted to photocurrent.
 
     temp_cell : numeric
         The average cell temperature of cells within a module in C.
 
-    alpha_isc : float
+    alpha_sc : float
         The short-circuit current temperature coefficient of the
-        module in units of 1/C.
-
-    module_parameters : dict
-        Parameters describing PV module performance at reference
-        conditions according to DeSoto's paper. Parameters may be
-        generated or found by lookup. For ease of use,
-        retrieve_sam can automatically generate a dict based on the
-        most recent SAM CEC module
-        database. The module_parameters dict must contain the
-        following 5 fields:
-
-            * a_ref - modified diode ideality factor parameter at
-              reference conditions (units of eV), a_ref can be calculated
-              from the usual diode ideality factor (n),
-              number of cells in series (Ns),
-              and cell temperature (Tcell) per equation (2) in [1].
-            * I_L_ref - Light-generated current (or photocurrent)
-              in amperes at reference conditions. This value is referred to
-              as Iph in some literature.
-            * I_o_ref - diode reverse saturation current in amperes,
-              under reference conditions.
-            * R_sh_ref - shunt resistance under reference conditions (ohms).
-            * R_s - series resistance under reference conditions (ohms).
+        module in units of A/C.
```

Comment:
C could be confused with Coulombs. Perhaps degC is preferable.

Reply:
I think the unit is clear in the context of the docstring.

Reply:
Amps/Coulomb = 1/second

Reply:
I'd like to leave the temperature unit on the temperature coefficients as C, because that's the unit used in IEC standards and datasheets.  I agree with @wholmgren about being clear in context.